### PR TITLE
Speed

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T24519

I have to leave now. I am not sure if this is the best way to do it. But this is working.

### CODE REVIEW

File: src/server.py [hunk 1]
```diff
@@ -52,6 +71,21 @@ def __bus_message_cb(self, unused_bus, message):
                             debug)
             self.pipeline.set_state(Gst.State.NULL)
             self.emit("error", error, debug)
+        elif message.type == Gst.MessageType.STATE_CHANGED:
+            if message.src != self.pipeline:
+                return
+            st = message.get_structure()
+            old_state = st.get_value("old-state")
+            new_state = st.get_value("new-state")
+            if old_state == Gst.State.READY and new_state == Gst.State.PAUSED:
```
Comment (ptomato) [inline src/server.py]: Shouldn't this be the other way around? When the new state is PLAYING?

File: src/server.py [hunk 2]
```diff
@@ -52,6 +71,21 @@ def __bus_message_cb(self, unused_bus, message):
...
+                if self.speed is not None:
```
Comment (ptomato) [inline src/server.py]: I'd suggest moving this up, before `message.get_structure()` and doing an early return:
```py
if self.speed is None:
    return
```

**Comment by ptomato** [comment]
We discussed an alternate approach to this on Slack.

**Comment by cfoch** [comment]
This is ready for review.

File: src/server.py [hunk 3]
```diff
@@ -86,6 +100,12 @@ def fade_out(self):
             return self.metadata["fade-out"]
         return None
 
+    @property
+    def speed(self):
```
Comment (ptomato) [inline src/server.py]: What does this property do, now that we have pitch and rate? I think we can remove it as well as the seek events, and just let soundtouch handle everything.

**Pull request merged**